core/src/main/scala/org/apache/spark/storage/TachyonBlockManager.scala

```diff
@@ -95,8 +94,23 @@ private[spark] class TachyonBlockManager() extends ExternalBlockManager with Log
   override def putBytes(blockId: BlockId, bytes: ByteBuffer): Unit = {
     val file = getFile(blockId)
     val os = file.getOutStream(WriteType.TRY_CACHE)
-    os.write(bytes.array())
-    os.close()
+    try {
+      os.write(bytes.array())
+      os.close()
```

Comment:
Should the `close()` happen in a `finally` block? here and below. This also doesn't cancel in case of a runtime exception, and also doesn't propagate the exception or even log it.